lib/t/feeds.ex

```diff
@@ -115,42 +105,107 @@ defmodule T.Feeds do
     {feed_profiles, feed_cursor}
   end
 
-  defp continue_feed(user_id, location, gender, gender_preferences, count) do
+  defp continue_feed(user_id, location, gender, feed_filter, count) do
+    %{genders: gender_preferences, min_age: min_age, max_age: max_age, distance: distance} =
+      feed_filter
+
     feeded = FeededProfile |> where(for_user_id: ^user_id) |> select([s], s.user_id)
 
     most_liked_count = count - div(count, 2)
 
     most_liked =
-      feed_profiles_q(user_id, gender, gender_preferences)
-      |> where([p], p.user_id not in subquery(feeded))
-      |> order_by(desc: :times_liked)
+      most_liked_q(user_id, gender, gender_preferences, feeded)
+      |> maybe_apply_age_filters(min_age, max_age)
+      |> maybe_apply_distance_filter(location, distance)
       |> limit(^most_liked_count)
-      |> select([p], {p, distance_km(^location, p.location)})
       |> Repo.all()
 
-    filter_out_ids = Enum.map(most_liked, fn {p, _} -> p.user_id end)
+    filter_out_ids = Enum.map(most_liked, fn p -> p.user_id end)
 
     most_recent_count = count - length(most_liked)
 
     most_recent =
-      feed_profiles_q(user_id, gender, gender_preferences)
-      |> where([p], p.user_id not in subquery(feeded))
-      |> where([p], p.user_id not in ^filter_out_ids)
-      |> order_by(desc: :last_active)
+      most_recent_q(user_id, gender, gender_preferences, feeded, filter_out_ids)
+      |> maybe_apply_age_filters(min_age, max_age)
+      |> maybe_apply_distance_filter(location, distance)
       |> limit(^most_recent_count)
-      |> select([p], {p, distance_km(^location, p.location)})
       |> Repo.all()
 
     most_liked ++ most_recent
   end
 
+  defp most_liked_q(user_id, gender, gender_preferences, feeded) do
+    feed_profiles_q(user_id, gender, gender_preferences)
+    |> where([p], p.user_id not in subquery(feeded))
+    |> order_by(desc: :times_liked)
+  end
+
+  defp most_recent_q(user_id, gender, gender_preferences, feeded, filter_out_ids) do
+    feed_profiles_q(user_id, gender, gender_preferences)
+    |> where([p], p.user_id not in subquery(feeded))
+    |> where([p], p.user_id not in ^filter_out_ids)
+    |> order_by(desc: :last_active)
+  end
+
+  defp maybe_apply_age_filters(query, min_age, max_age) do
+    query
+    |> maybe_apply_min_age_filer(min_age)
+    |> maybe_apply_max_age_filer(max_age)
+  end
+
+  defp maybe_apply_min_age_filer(query, min_age) do
+    case min_age do
```

Comment:
```elixir
if min_age do
  where(...)
else
  query
end
```
anything more is pure complication

Reply:
why not?

Reply:
what I wrote above is easy to understand

Reply:
please do the approach above, it's easier to understand. applying `where` doesn't hurt, applying `where` only when there is a filter in db is correct